go/inst/oracle_gtid_set_entry.go

```diff
@@ -28,47 +28,250 @@ var (
+	}
+
+	// Tagged ranges are added in the end of the Gtid_set
+	for _, v := range this.TaggedIv {
+		res += ":" + v.Tag
+		for _, iv := range v.Interval {
+			res += ":" + iv
+		}
+	}
+	return res
 }
 
-// String returns a user-friendly string representation of this entry
+/*
+String returns a user-friendly individual string representation of the gtid set
+
+Example:
+Explode of the GTID set "48ebed33-0d12-11ef-a3ec-ac198e4551c8:1-3:7:tag1:1-2:10-12:tag2:74-75:78:81"
+shall return the following
+
+48ebed33-0d12-11ef-a3ec-ac198e4551c8:1
+48ebed33-0d12-11ef-a3ec-ac198e4551c8:2
+48ebed33-0d12-11ef-a3ec-ac198e4551c8:3
+48ebed33-0d12-11ef-a3ec-ac198e4551c8:7
+48ebed33-0d12-11ef-a3ec-ac198e4551c8:tag1:1
+48ebed33-0d12-11ef-a3ec-ac198e4551c8:tag1:2
+48ebed33-0d12-11ef-a3ec-ac198e4551c8:tag1:10
+48ebed33-0d12-11ef-a3ec-ac198e4551c8:tag1:11
+48ebed33-0d12-11ef-a3ec-ac198e4551c8:tag1:12
+48ebed33-0d12-11ef-a3ec-ac198e4551c8:tag2:74
+48ebed33-0d12-11ef-a3ec-ac198e4551c8:tag2:75
+48ebed33-0d12-11ef-a3ec-ac198e4551c8:tag2:78
+48ebed33-0d12-11ef-a3ec-ac198e4551c8:tag2:81
+*/
 func (this *OracleGtidSetEntry) Explode() (result [](*OracleGtidSetEntry)) {
-	intervals := strings.Split(this.Ranges, ":")
-	for _, interval := range intervals {
-		if submatch := multiValueInterval.FindStringSubmatch(interval); submatch != nil {
-			intervalStart, _ := strconv.Atoi(submatch[1])
-			intervalEnd, _ := strconv.Atoi(submatch[2])
-			for i := intervalStart; i <= intervalEnd; i++ {
-				result = append(result, &OracleGtidSetEntry{UUID: this.UUID, Ranges: fmt.Sprintf("%d", i)})
+
+	// Appends the default interval to the result
+	var AppendDefaultInterval = func(this *OracleGtidSetEntry) {
+		intervals := strings.Split(this.DefaultIv, ":")
+		for _, interval := range intervals {
+			// Multi-value interval
+			if submatch := multiValueInterval.FindStringSubmatch(interval); submatch != nil {
+				intervalStart, _ := strconv.Atoi(submatch[1])
+				intervalEnd, _ := strconv.Atoi(submatch[2])
+				for i := intervalStart; i <= intervalEnd; i++ {
+					result = append(result, &OracleGtidSetEntry{UUID: this.UUID, DefaultIv: fmt.Sprintf("%d", i)})
+				}
+			} else if submatch := singleValueInterval.FindStringSubmatch(interval); submatch != nil {
+				// Single-value interval
+				result = append(result, &OracleGtidSetEntry{UUID: this.UUID, DefaultIv: interval})
+			}
+		}
+	}
+
+	// Appends tagged intervals to the result
+	var AppendTaggedInterval = func(tag *string, interval string) {
+
+		intervals := strings.Split(interval, ":")
+		for _, interval := range intervals {
+
+			// Multi-value interval
+			if submatch := multiValueInterval.FindStringSubmatch(interval); submatch != nil {
+
+				intervalStart, _ := strconv.Atoi(submatch[1])
+				intervalEnd, _ := strconv.Atoi(submatch[2])
+				for i := intervalStart; i <= intervalEnd; i++ {
+					var ti TagInterval
+					ti.Tag = *tag
+					ti.Interval = append(ti.Interval, fmt.Sprintf("%d", i))
+
+					var taggedIv []TagInterval
```

Comment:
Won't the following work?
`taggedIv := []TagInterval{ti}`

Reply:
It works. Fixed it.